app/fileReceiver.js

```diff
@@ -51,6 +51,28 @@ export default class FileReceiver extends Nanobus {
     this.state = 'ready';
   }
 
+  async streamToArrayBuffer(stream) {
+    const reader = stream.getReader();
+    const chunks = [];
+    let length = 0;
+
+    let state = await reader.read();
+    while (!state.done) {
+      chunks.push(state.value);
+      length += state.value.length;
+      state = await reader.read();
+    }
+
+    const result = new Int8Array(length);
```

Comment:
Should this be an unsigned `UInt8Array`?

Reply:
Yes because I want to concat all of the stream output, but ArrayBuffer doesn't have methods for direct manipulation. I can refactor this to loop through only once though.